cpp/include/cugraph/mtmg/resource_manager.hpp

```diff
@@ -130,29 +152,40 @@ class resource_manager_t {
                                                               ncclUniqueId instance_manager_id,
                                                               size_t n_streams = 16) const
   {
-    std::for_each(
-      ranks_to_include.begin(), ranks_to_include.end(), [local_ranks = local_rank_map_](int rank) {
-        CUGRAPH_EXPECTS(local_ranks.find(rank) != local_ranks.end(),
-                        "requesting inclusion of an invalid rank");
-      });
+    std::vector<int> local_ranks_to_include;
+
+    std::for_each(ranks_to_include.begin(),
+                  ranks_to_include.end(),
+                  [&local_ranks  = local_rank_map_,
+                   &remote_ranks = remote_rank_set_,
+                   &local_ranks_to_include](int rank) {
+                    if (local_ranks.find(rank) == local_ranks.end()) {
+                      CUGRAPH_EXPECTS(remote_ranks.find(rank) != remote_ranks.end(),
+                                      "requesting inclusion of an invalid rank");
+                    } else {
+                      local_ranks_to_include.push_back(rank);
+                    }
+                  });
```

Comment:
I think using `std::copy_if` with `std::back_inserter` better documents the intention.

Reply:
I will look at this change.

Reply:
Updated in next push.